app/controllers/journeys/nics/Class4DivingExemptController.scala

```diff
@@ -58,8 +59,8 @@ class Class4DivingExemptController @Inject() (override val messagesApi: Messages
       .fold(
         formErrors => Future.successful(BadRequest(view(formErrors, taxYear, request.userType, mode, businesses))),
         answer => {
-          val filteredAnswer = if (answer.head.value.isEmpty) List.empty else answer
-          service.persistAnswerAndRedirect(page, nationalInsuranceContributions, request, filteredAnswer, taxYear, mode)
+          val filteredAnswer = if (answer.head.value.isEmpty) List(classFourOtherExemption) else answer
```

Comment:
not your code, but answer.head may throw an exception, it would be safer to call headOption instead